analyze.py

```diff
@@ -247,34 +268,52 @@ def contributorTimeline(self, name):
         contributorDataSortedByNMessages,
     )),
     (contributorNames,) = list(map(
-        lambda _contributorData: _contributorData[0][:11] + "..." if len(_contributorData[0]) > 14 else _contributorData[0],
+        lambda _contributorData: _contributorData[0][:11]
+        + "..." if len(_contributorData[0]) > 14 else _contributorData[0],
         contributorDataSortedByNMessages,
     )),
 
     # Contribution timeline
     if args.timeline:
         figure = plt.figure()
 
-        for contributorData in contributorDataSortedByNMessages[:20]:
+        for contributorData in contributorDataSortedByNMessages[:10]:
             contributor = contributorData[0]
             nMessages = contributorData[1]
             nWords = contributorData[2]
             timeline = contributorData[3]
 
+            x = timeline.keys()
+            y = timeline.values()
+            x = numpy.array(list(x))
+            y = numpy.array(list(y))
+
+            if args.start_date is not None:
+                for i in range(len(x)):
```

Comment:
I'd prefer a `list(filter(lambda key,value: key > args.start_date.date(), timeline))` type construct
and plotting the values of that (optionally minus their first)

less code is easier to read & maintain